openmdao/recorders/sqlite_reader.py

```diff
@@ -309,16 +321,27 @@ def list_sources(self):
         if self._format_version >= 2 and self._problem_cases.count() > 0:
             sources.extend(self._problem_cases.list_sources())
 
-        return sources
+        if not out_stream:
+            return sources
+        else:
+            if out_stream is _DEFAULT_OUT_STREAM:
+                out_stream = sys.stdout
+
+            if out_stream:
+                for source in sources:
+                    out_stream.write('{}\n'.format(source))
```

Comment:
I think this should always have a return value (`sources`) regardless of the out_stream, similar to the `list_` functions on System (and in keeping with the docstring)